core/tests/puppeteer-acceptance-tests/functions/is-element-clickable.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility function to check if an element is clickable.
+ */
+
+export default function isElementClickable(element: Element): boolean {
+  /**
+   * This function gets the the overlapping element if any by checking the
+   * element that is present in the center of the target element's position.
+   */
+  const getOverlappingElement = (
+    element: Element,
+    parent: ShadowRoot | Document = document
+  ): Element | null => {
+    const elementDimensions = element.getBoundingClientRect();
+    const x = elementDimensions.left + element.clientWidth / 2;
+    const y = elementDimensions.top + element.clientHeight / 2;
+
+    return parent.elementFromPoint(x, y);
+  };
+
+  /**
+   * This function also gets the overlapping element if any at the center
```

Comment:
Can you have this docstring make it clear to devs when exactly they should use getOverlappingRect vs getOverlappingElement?

Reply:
Done